testutils/lint: crlfmt's diff output is broken

Since 40c2b39 lint errors have extra newlines in them. This doesn't work well for crlfmt's output, which is supposed to be a diff. The extra newlines make it very hard to parse.

```
    --- FAIL: TestLint/TestCrlfmt (4.24s)
    	lint_test.go:728: 
    		/Users/benesch/go/src/github.com/cockroachdb/cockroach/pkg/storage/batcheval/cmd_end_transaction.go:1035
    	lint_test.go:728: 
    		-func changeReplicasTrigger(
    	lint_test.go:728: 
    		-	ctx context.Context,
    	lint_test.go:728: 
    		-	rec EvalContext,
    	lint_test.go:728: 
    		-	batch engine.Batch,
    	lint_test.go:728: 
    		-	change *roachpb.ChangeReplicasTrigger,
    	lint_test.go:728: 
    		-) result.Result {
    	lint_test.go:728: 
    		+func changeReplicasTrigger(
    	lint_test.go:728: 
    		+	ctx context.Context, rec EvalContext, batch engine.Batch, change *roachpb.ChangeReplicasTrigger,
    	lint_test.go:728: 
    		+) result.Result {
    	lint_test.go:728: 
    	lint_test.go:728: 
```

--- a/pkg/testutils/lint/lint_test.go
+++ b/pkg/testutils/lint/lint_test.go
@@ -724,10 +724,15 @@ func TestLint(t *testing.T) {
 			t.Fatal(err)
 		}
 
+		var buf bytes.Buffer
 		if err := stream.ForEach(filter, func(s string) {
-			t.Errorf("\n%s", s)
+			fmt.Fprintln(&buf, s)
 		}); err != nil {
 			t.Error(err)
+		}
+		errs := buf.String()
+		if len(errs) > 0 {
+			t.Errorf("\n%s", errs)
 		}
 
 		if err := cmd.Wait(); err != nil {

